pkg/vm/builtins/ecop.go

```diff
@@ -0,0 +1,234 @@
+package builtins
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	mem "github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	f "github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+	"github.com/holiman/uint256"
+)
+
+const EcOpName = "ec_op"
+const cellsPerEcOp = 7
+const inputCellsPerEcOp = 5
+
+var feltThree f.Element = f.Element(
+	[]uint64{
+		18446744073709551521,
+		18446744073709551615,
+		18446744073709551615,
+		576460752303421872,
+	})
+
+type EcOp struct{}
+
+func (e *EcOp) String() string {
+	return EcOpName
+}
+
+func (e *EcOp) CheckWrite(segment *mem.Segment, offset uint64, value *mem.MemoryValue) error {
+	return nil
+}
+
+func (e *EcOp) InferValue(segment *mem.Segment, offset uint64) error {
+	// get the current slot index and verify it is an output cell
+	ecopIndex := offset % cellsPerEcOp
+	if ecopIndex < inputCellsPerEcOp {
+		return errors.New("cannot infer value from input cell")
+	}
+
+	// gather the input cells
+	inputOff := offset - ecopIndex
+	inputs := [5]mem.MemoryValue{
+		segment.Peek(inputOff),
+		segment.Peek(inputOff + 1),
+		segment.Peek(inputOff + 2),
+		segment.Peek(inputOff + 3),
+		segment.Peek(inputOff + 4),
+	}
+
+	// assert all values are known
+	for i := range inputs {
+		if !inputs[i].Known() {
+			return fmt.Errorf(
+				"cannot infer value: input value at offset %d is unknown", inputOff+uint64(i),
+			)
+		}
+	}
+
+	// unwrap the values as felts
+	inputsFelt := [5]*f.Element{}
+	for i := range inputs {
+		felt, err := inputs[i].FieldElement()
+		if err != nil {
+			return err
+		}
+		inputsFelt[i] = felt
+	}
+
+	// Note: the python vm has an upper limit on the size of `m`(the fifth input)  but
+	// since it is always maxout at 2**252, I see no point on adding a check
+	// for it for now
+
+	// alpha and beta are paremeters required by the curve
+	// extracted from pedersen_params.json in https://github.com/starkware-libs/cairo-lang
+	alpha := f.One()
```

Comment:
Let's move these to the util package cause ecdsa builtin also requires the same values